NetKVM/Common/ParaNdis_TX.cpp

```diff
@@ -775,6 +823,13 @@ void CNB::ReleaseResources()
         NdisMFreeNetBufferSGList(m_Context->DmaHandle, m_SGL, m_NB);
         m_SGL = nullptr;
     }
+#if MAX_FRAGMENTS_IN_ONE_NB
+    if (m_ExtraNBStorage)
+    {
+        NdisFreeMemoryWithTagPriority(m_Context->MiniportHandle, m_ExtraNBStorage, PARANDIS_MEMORY_TAG);
+        m_ExtraNBStorage = nullptr;
```

Comment:
@ybendito ，I've added returnpages in both AllocateAndFillCopySGL (which will return resources in case of internal failure) and CTXVirtQueue::ReleaseOneBuffer. My understanding is that this should suffice, but I'm not sure if there's anything I might have overlooked.

Reply:
@zjmletang   Probably you're right. Did you run your tests with driver verifier?

Reply:
@ybendito ，Yes, I've run an overnight test, about 12 hours or so. There were no anomalies.